### PR TITLE
Fix #240: No Ports exposed inside Deployment in case of Zero Config Dockerfile Mode

### LINKED ISSUE #240: No Ports exposed inside Deployment in case of Zero Config Dockerfile Mode
While modifying integration test PR(https://github.com/jkubeio/jkube-integration-tests/pull/45), I noticed that generated resource manifest seem to be missing ports. Our Enricher API relies only on `ImageConfiguration` field to fetch ports:
https://github.com/eclipse/jkube/blob/07477a232e9ed4d3441622e43fda8f1c9d8466bf/jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/kit/enricher/handler/ContainerHandler.java#L164-L170

Here is an example of generated Deployment:
```
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    jkube.io/git-url: git@github.com:jkubeio/jkube-integration-tests.git
    jkube.io/git-commit: abf5c3014bd084a15c59e8b4fd19becf08b2bc0e
    jkube.io/git-branch: pr/simple-dockerfile-test
  labels:
    app: spring-boot-zero-config-dockerfile
    provider: jkube
    version: 0.0.0-SNAPSHOT
    group: org.eclipse.jkube.integration-tests
  name: spring-boot-zero-config-dockerfile
spec:
  replicas: 1
  revisionHistoryLimit: 2
  selector:
    matchLabels:
      app: spring-boot-zero-config-dockerfile
      provider: jkube
      group: org.eclipse.jkube.integration-tests
  template:
    metadata:
      annotations:
        jkube.io/git-url: git@github.com:jkubeio/jkube-integration-tests.git
        jkube.io/git-commit: abf5c3014bd084a15c59e8b4fd19becf08b2bc0e
        jkube.io/git-branch: pr/simple-dockerfile-test
      labels:
        app: spring-boot-zero-config-dockerfile
        provider: jkube
        version: 0.0.0-SNAPSHOT
        group: org.eclipse.jkube.integration-tests
    spec:
      containers:
      - env:
        - name: KUBERNETES_NAMESPACE
          valueFrom:
            fieldRef:
              fieldPath: metadata.namespace
        image: integration-tests/spring-boot-zero-config-dockerfile:latest
        imagePullPolicy: IfNotPresent
        name: integration-tests-spring-boot-zero-config-dockerfile
        securityContext:
          privileged: false
```

As you can see, there is no port being provided inside `.spec.template.spec.containers[0]`. Sadly [DefaultControllerEnricher](https://github.com/eclipse/jkube/blob/master/jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/DefaultControllerEnricher.java#L76) doesn't have any parameter to configure ports:

https://github.com/eclipse/jkube/blob/07477a232e9ed4d3441622e43fda8f1c9d8466bf/jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/DefaultControllerEnricher.java#L76-L83

Maybe we can add an option just like `DefaultServiceEnricher`:
https://github.com/eclipse/jkube/blob/07477a232e9ed4d3441622e43fda8f1c9d8466bf/jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/DefaultServiceEnricher.java#L85-L100

### PR DESCRIPTION
**Pull request by rohanKanojia**
Fix #240 
Add ports to ImageConfiguration generated in DockerFileUtil.createSimpleDockerfileConfig
by parsing contents of Dockerfile

## Type of change

 - [X] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [X] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [X] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [X] I have implemented unit tests to cover my changes
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [x] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [X] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: jkube-kit/build/api/src/main/java/org/eclipse/jkube/kit/build/api/helper/DockerFileUtil.java [hunk 1]
```diff
@@ -312,4 +317,23 @@ private static String trimSurroundingQuotes(String argStringValue) {
         }
         return argStringValue;
     }
+
+    static List<String> getPortsFromDockerFile(File dockerFile) {
```
Comment (manusa) [inline jkube-kit/build/api/src/main/java/org/eclipse/jkube/kit/build/api/helper/DockerFileUtil.java:321]: nit: Please make it consistent, rename to `extractPorts`

File: jkube-kit/build/api/src/main/java/org/eclipse/jkube/kit/build/api/helper/DockerFileUtil.java [hunk 2]
```diff
@@ -312,4 +317,23 @@ private static String trimSurroundingQuotes(String argStringValue) {
...
+        Properties properties = new Properties();
+        try {
+            return getPortsFromDockerFileLines(extractLines(dockerFile, "EXPOSE", properties, null));
+        } catch (IOException ioException) {
+            throw new IllegalArgumentException("Error in reading Dockerfile", ioException);
+        }
+    }
+
+    static List<String> getPortsFromDockerFileLines(List<String[]> dockerLinesContainingExpose) {
```
Comment (manusa) [inline jkube-kit/build/api/src/main/java/org/eclipse/jkube/kit/build/api/helper/DockerFileUtil.java:330]: If this is visible for testing, make it private and test using the dependent method (should be easy enough).

Also, nit: rename it to be consistent: `extractPorts`

File: jkube-kit/build/api/src/test/resources/docker/Dockerfile_expose_ports [hunk 3]
```diff
@@ -0,0 +1,7 @@
+ARG VERSION=latest
+FROM openjdk:$VERSION
+COPY maven/target/docker-file-simple.jar /deployments/docker-file-simple.jar
+# Copying a file inside project root directory
+COPY maven/static-dir-in-project-root/my-file.txt /deployments/my-file.txt
+EXPOSE 8080
```
Comment (manusa) [inline jkube-kit/build/api/src/test/resources/docker/Dockerfile_expose_ports:6]: Use a corner-case scenario to validate it works as expected:
- Add several EXPOSE lines (top of the document, bottom, in the middle lines).
- Add expose using all possible formats: `EXPOSE 80`, `EXPOSE 80/http`, `EXPOSE 99/udp`, `EXPOSE 8080`...
- Add duplicate EXPOSE values

**Review by manusa — APPROVED** [review]
